cmd/test-image/main.go

```diff
@@ -0,0 +1,20 @@
+package main
+
+import (
+	"github.com/docker/libnetwork/cmd/test-image/exec"
+	"github.com/docker/libnetwork/cmd/test-image/icmp"
+	"github.com/docker/libnetwork/cmd/test-image/server"
+	"github.com/docker/libnetwork/cmd/test-image/tcp"
+)
+
+func main() {
+	s := server.NewTestServer()
+	s.Init()
+	// enable ICMP endpoints
+	icmp.Init(s)
+	tcp.Init(s)
+	exec.Init(s)
+
+	s.Start("0.0.0.0", 2000)
+	select {}
```

Comment:
Will this abort abnormally when the testserver goroutine exits?  (i.e. with a deadlock detection condition?)

Reply:
this is an intentional deadlock :D